test/unit/chtransfer_test.go

```diff
@@ -80,6 +83,203 @@ func TestByCustomerForwardSuccess(t *testing.T) {
+	}
+	itemsJSON, err = json.Marshal(items)
+	require.NoError(t, err)
+
+	err = owner.RawSignedInvokeWithErrorReturned("cc", "channelMultiTransferByAdmin", uuid.NewString(), "VT", user1.Address(), string(itemsJSON))
+	require.EqualError(t, err, cctransfer.ErrInvalidTokenAlreadyExists.Error())
+
+	items = make([]core.TransferItem, 0, 100)
+	for i := 0; i < 101; i++ {
+		itemToken := fmt.Sprintf("CC_%d", i)
+		items = append(items, core.TransferItem{Token: itemToken, Amount: new(big.Int).SetInt64(1)})
+		user1.AddTokenBalance("cc", itemToken, 1)
+	}
+	itemsJSON, err = json.Marshal(items)
+	require.NoError(t, err)
+	err = owner.RawSignedInvokeWithErrorReturned("cc", "channelMultiTransferByAdmin", uuid.NewString(), "VT", user1.Address(), string(itemsJSON))
+	require.EqualError(t, err, "invalid argument transfer items count found 101 but expected from 1 to 100")
+}
+
+func TestMultiTransferByCustomerForwardSuccess(t *testing.T) {
+	ledger := mock.NewLedger(t)
+	owner := ledger.NewWallet()
+
+	ccConfig := makeBaseTokenConfig("CC Token", "CC", 8,
+		owner.Address(), "", "", "", nil)
+	initMsg := ledger.NewCC("cc", &token.BaseToken{}, ccConfig)
+	require.Empty(t, initMsg)
+
+	vtConfig := makeBaseTokenConfig("VT Token", "VT", 8,
+		owner.Address(), "", "", "", nil)
+	initMsg = ledger.NewCC("vt", &token.BaseToken{}, vtConfig)
+	require.Empty(t, initMsg)
+
+	user1 := ledger.NewWallet()
+	user1.AddTokenBalance("cc", "CC_1", 1000)
+	user1.AddTokenBalance("cc", "CC_2", 2000)
+
+	id := uuid.NewString()
+
+	items := []core.TransferItem{
+		{Token: "CC_1", Amount: new(big.Int).SetInt64(450)},
+		{Token: "CC_2", Amount: new(big.Int).SetInt64(900)},
+	}
+
+	itemsJSON, err := json.Marshal(items)
+	require.NoError(t, err)
+
+	err = user1.RawSignedInvokeWithErrorReturned("cc", "channelMultiTransferByCustomer", id, "VT", string(itemsJSON))
+	require.NoError(t, err)
+	cct := user1.Invoke("cc", "channelTransferFrom", id)
+
+	_, _, err = user1.RawChTransferInvokeWithBatch("vt", "createCCTransferTo", cct)
+	require.NoError(t, err)
+	ledger.WaitChTransferTo("vt", id, time.Second*5)
+	_ = user1.Invoke("vt", "channelTransferTo", id)
+
+	_, _, err = user1.RawChTransferInvoke("cc", "commitCCTransferFrom", id)
+	require.NoError(t, err)
+
+	_, _, err = user1.RawChTransferInvoke("vt", "deleteCCTransferTo", id)
+	require.NoError(t, err)
+
+	_, _, err = user1.RawChTransferInvoke("cc", "deleteCCTransferFrom", id)
+	require.NoError(t, err)
+
+	err = user1.InvokeWithError("cc", "channelTransferFrom", id)
+	require.Error(t, err)
+	err = user1.InvokeWithError("vt", "channelTransferTo", id)
+	require.Error(t, err)
+
+	//user1.BalanceShouldBe("cc", 550)
+	user1.AllowedBalanceShouldBe("vt", "CC_1", 450)
+	user1.AllowedBalanceShouldBe("vt", "CC_2", 900)
+	//user1.CheckGivenBalanceShouldBe("vt", "VT", 0)
+	//user1.CheckGivenBalanceShouldBe("vt", "CC", 0)
+	//user1.CheckGivenBalanceShouldBe("cc", "CC", 0)
+	//user1.CheckGivenBalanceShouldBe("cc", "VT", 450)
```

Comment:
unnecessary comments should be deleted or uncommented

Reply:
fixed